### ISSUE TITLE
IntPtr type?

### ISSUE THREAD

**Issue by sylveon** [MicrosoftDocs/winrt-related#211]
Is there a type that represents a C# IntPtr or C++ std::intptr_t? I need to pass a HWND in my constructor so I thought of using something like IntPtr but I can't seem to find one. Or is there a way to directly pass an HWND through WinRT? The ability to mix classic IDL with modern IDL seems interesting since classic IDL supports it, but I'm not sure if I can use that for a constructor.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2dacbdf2-172f-dd41-829f-94295cd0c621
* Version Independent ID: b511accc-e0b1-9b30-d044-231d5204024d
* Content: [Introduction to Microsoft Interface Definition Language 3.0 - Windows UWP applications](https://docs.microsoft.com/en-us/uwp/midl-3/intro)
* Content Source: [winrt-related-src/midl-3/intro.md](https://github.com/MicrosoftDocs/winrt-related/blob/docs/winrt-related-src/midl-3/intro.md)
* Product: **uwp**
* Technology: **winrt-reference**
* GitHub Login: @stevewhims
* Microsoft Alias: **stwhi**

**Comment by stevewhims** [comment]
Hi, @sylveon, thanks for filing this issue.

For passing a HWND to a Windows Runtime object, I'd suggest implementing [IInitializeWithWindow](https://docs.microsoft.com/windows/win32/api/shobjidl_core/nn-shobjidl_core-iinitializewithwindow). A less elegant approach is to smuggle a HWND (or indeed any kind of HANDLE) by means of a UInt64 parameter.

-Steve

**Comment by sylveon** [comment]
I did try implementing `IInitializeWithWindow` shortly after filling this question, but faced an issue where the XAML compiler's codegen did an explicit template specialization that causes compiler errors:

My code:
![image](https://user-images.githubusercontent.com/6440374/94216936-ba34ab80-feae-11ea-8973-78ae52a0f3e1.png)

ColorPickerPage.xaml.g.hpp: 
![image](https://user-images.githubusercontent.com/6440374/94216968-cf113f00-feae-11ea-9e26-e4a91fa49a99.png)

Since ColorPickerPage inherits from `ColorPickerPageT<ColorPickerPage, IInitializeWithWindow>` rather than `ColorPickerPageT<ColorPickerPage>` any attempt to cast back to the derived type caused a compiler failure. Even if the code didn't do that, I would have hit linker errors. Since this is automatically generated, there is no easy resolution I can do on my side.

I tried a hack by doing `#define ColorPickerPage ColorPickerPage, IInitializeWithWindow`, while it does work it's very fragile and I would rather not ship that code.

Of course this works fine on standard WinRT objects, and I was going to fill an issue on the microsoft-ui-xaml repo later.
